cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/ObjectMappedTableDefinition.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.dataset2.lib.table;
+
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.data.schema.UnsupportedTypeException;
+import co.cask.cdap.api.dataset.DatasetAdmin;
+import co.cask.cdap.api.dataset.DatasetDefinition;
+import co.cask.cdap.api.dataset.DatasetProperties;
+import co.cask.cdap.api.dataset.DatasetSpecification;
+import co.cask.cdap.api.dataset.lib.AbstractDatasetDefinition;
+import co.cask.cdap.api.dataset.lib.KeyValueTable;
+import co.cask.cdap.api.dataset.lib.ObjectMappedTable;
+import co.cask.cdap.api.dataset.lib.ObjectStore;
+import co.cask.cdap.api.dataset.table.Table;
+import co.cask.cdap.internal.io.SchemaTypeAdapter;
+import co.cask.cdap.internal.io.TypeRepresentation;
+import com.google.common.base.Preconditions;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * DatasetDefinition for {@link ObjectMappedTableDataset}.
+ */
+public class ObjectMappedTableDefinition extends AbstractDatasetDefinition<ObjectMappedTable, DatasetAdmin> {
+
+  private static final Gson GSON = new Gson();
+  private final DatasetDefinition<? extends Table, ?> tableDef;
+
+  public ObjectMappedTableDefinition(String name, DatasetDefinition<? extends Table, ?> tableDef) {
+    super(name);
+    Preconditions.checkArgument(tableDef != null, "Table definition is required");
+    this.tableDef = tableDef;
+  }
+
+  @Override
+  public DatasetSpecification configure(String instanceName, DatasetProperties properties) {
+    Preconditions.checkArgument(properties.getProperties().containsKey("type"));
```

Comment:
want to validate the schema, and this is an earlier place than when somebody calls getDataset.  The schema cant be generated from the Type at this point because this gets called in the system context. If the Type is in the user code, it will not be available here.